### ISSUE TITLE
Bridge has low speed while receiving TCP packet on QEMU vm

### ISSUE THREAD

**Issue by howelite** [luigirizzo/netmap#257]
Hello,
I am using netmap bridge on a QEMU vm, running CentOS 7 and has virtio_net NIC driver.I use unpatched driver and disabled tx-checksumming and gso/gro/tso. Then i bridging NIC and host stack using netmap bridge, i run iperf3 on this host between another host(not use netmap), the host that using netmap has low speed while receiving TCP packet, but normal on UDP. I don't know why, Could you give me some advice?

**Comment by vmaffione** [comment]
Hi,
  I assume by "netmap bridge" you mean an instance of the VALE switch.

  First of all, make sure to disable the offloadings (chksum + tso/gso/gro/lro) both inside the VM (on the virtio_net interface) and on the host (on the phyisical NIC that you attach to the VALE switch.

Then, consider that netmap is designed to provide an high performance gain with netmap applications (that are able to transmit/receive in batch, possibly with zero copy). It does not usually provide a real gain if you are using TCP/IP socket applications, specially if those applications run on Linux and use large TSO packets (like netperf/iperf does by default); it sometimes provides some gain in non-Linux virtualization environments (e.g. FreeBSD), because an highly optimized virtio solution may not be available for those environments.

In other words, you need to make sure that what you want to do matches the netmap use-cases.

**Comment by vmaffione** [comment]
This seems to be a matter of unapproriate usecases,not a problem of netmap itself
